### ISSUE TITLE
Problem getting mypy extension to pyls to work

### ISSUE THREAD

**Issue by jorgsk** [neovim/nvim-lspconfig#353]
Thanks for the amazing plug in and work!

I've got the following configuration:

```
lua << EOF
local on_attach_vim = function(client)
  require'completion'.on_attach(client)
  require'diagnostic'.on_attach(client)
end
require'nvim_lsp'.pyls.setup{on_attach = on_attach_vim,
                             settings = {
                               pyls = {
                                  plugins = {
                                       pyls_mypy = {enabled = true}
                                       }
                                   }
                                 }
                              }
EOF
```
where the pyls_mypy configuration is assumed from it's documentation: https://github.com/tomv564/pyls-mypy
I get pyls to work but not this extension.

I've tried to find out if pyls is aware of mypy or if the problem lies on the configuration side, but I haven't found any way to get started. Any help to get started debugging is appreciated. I'm not sure if it's related, but neovim does not draw any underlines for the diagnostics, but I get the popup. See the attached screenshot for how pyls reports about missing empty line after class definition but does not complain about the type problem. Running mypy on the command line gives the right error commands.
![no_mypy](https://user-images.githubusercontent.com/761677/93333439-996fb480-f823-11ea-8a41-393a00ea549a.png)

**Comment by lithammer** [comment]
How did you install pyls? And is pyls-mypy also installed in the same place? You can also use this command to get the location of the log file:

    :lua print(vim.lsp.get_log_path())

**Comment by jorgsk** [comment]
I installed pyls and pyls-mypy with pip, although I'm working in a conda enviroment so they end up under ~/miniconda3/lib/python3.7/site-packages/.

Thanks for the log path! The first entry was:
```
[ ERROR ] 2020-09-16T10:19:19+0200 ] /snap/nvim/323/usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:428 ]	"rpc"	"pyls"	"stderr"	"2020-09-16 10:19:19,209 UTC - WARNING - pyls.config.config - Failed to load pyls entry point 'pyls_mypy': No module named 'future'\n"
```
Seems that the future module is a dependency for pyls_mypy that was not installed automatically by pip. I installed it, which solved the issue. Hopefully this will help someone else who encounters the same issue.
